.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,66 @@
+name: Build and publish artifacts
+
+on: push
+
+jobs:
+  build-artifacts:
+    name: Build Python packages
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+        
+    - name: Install pypa/build
+      run: >-
+        python -m
+        pip install
+        build
+        --user
+    - name: Build a binary wheel and a source tarball
+      run: >-
+        python -m
+        build
+        --sdist
+        --wheel
+        --outdir dist/
+        .
+
+    - name: Archive build artifacts
```

Comment:
Uploading built artifacts to GH Actions so that we can use it in the publish job, and also so that we can download built artifacts if we need to.